ebean-datasource/src/test/java/io/ebean/datasource/test/FactoryTest.java

```diff
@@ -49,4 +52,39 @@ public void dataSourceFactory_get_createPool() throws Exception {
     }
 
   }
+
+
+  @Test
+  public void testPreparedStatement() throws Exception {
+
+    DataSourceConfig config = new DataSourceConfig();
+    config.setDriver("org.h2.Driver");
+    config.setUrl("jdbc:h2:mem:tests");
+    config.setUsername("sa");
+    config.setPassword("");
+
+    DataSourcePool pool = DataSourceFactory.create("test", config);
+    String sql = "select * from information_schema.settings where setting_name != ?";
+    try (Connection connection = pool.getConnection()) {
+      try (PreparedStatement stmt = connection.prepareStatement(sql)) {
+        stmt.executeQuery();
+        Assertions.fail("Not expected");
+      } catch (SQLException e) {
+        // expected: Parameter "#1" not set
+      }
+
+      try (PreparedStatement stmt = connection.prepareStatement(sql)) {
+        stmt.setString(1, "foo");
+        ResultSet rs = stmt.executeQuery();
+        rs.close();
```

Comment:
a prepared statement, that is put back to the pool keeps the result open (and may be also other objects like ResultSetMetaData). This might consume more resources than wanted.

Possible workarounds:
- call `while (pstmt.getMoreResults(Statement.CLOSE_ALL_RESULTS)) {}`
- ExtendedPreparedStatement could "remember" the last returend ResultSet and we close this resultSet, when the connection is returned to the pool.

Note: this is only "not nice behavour" which was discovered while analyzing the bind param problem.